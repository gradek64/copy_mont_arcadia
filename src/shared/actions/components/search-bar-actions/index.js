export {
  toggleSearchBar,
  closeProductsSearchBar,
  trackSearchBarSelected,
} from './search-bar-actions'

export { toggleProductsSearchBar, submitSearch } from './search-bar-thunks'
