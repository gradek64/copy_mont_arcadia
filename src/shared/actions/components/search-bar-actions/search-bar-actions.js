import {
  sendAnalyticsClickEvent,
  GTM_CATEGORY,
  GTM_ACTION,
} from '../../../analytics'

export function toggleSearchBar() {
  return {
    type: 'TOGGLE_PRODUCTS_SEARCH_BAR',
  }
}

export function closeProductsSearchBar() {
  return {
    type: 'CLOSE_PRODUCTS_SEARCH_BAR',
  }
}

export function trackSearchBarSelected() {
  return sendAnalyticsClickEvent({
    category: GTM_CATEGORY.SEARCH_SELECTED,
    action: GTM_ACTION.SEARCH_BAR,
  })
}
