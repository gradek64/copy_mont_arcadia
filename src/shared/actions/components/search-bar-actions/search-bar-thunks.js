import { browserHistory } from 'react-router'
import { clearSortOptions } from '../sortSelectorActions'
import { clearRefinements } from '../refinementsActions'
import { resetForm } from '../../common/formActions'
import { clearInfinityPage } from '../../common/infinityScrollActions'
import { analyticsSearchClickEvent } from '../../../analytics/tracking/site-interactions'
import { trackSearchBarSelected, toggleSearchBar } from './search-bar-actions'
import { isSearchBarOpen } from '../../../selectors/productSearchSelectors'

export function toggleProductsSearchBar() {
  return (dispatch, getState) => {
    dispatch(toggleSearchBar())
    if (isSearchBarOpen(getState())) {
      dispatch(trackSearchBarSelected())
    }
  }
}

export function submitSearch(searchTerm) {
  return (dispatch, getState) => {
    if (searchTerm) {
      analyticsSearchClickEvent(searchTerm)
      if (getState().productsSearch.open) dispatch(toggleProductsSearchBar())
      dispatch(clearSortOptions())
      dispatch(clearRefinements())
      dispatch(clearInfinityPage())

      // Free text search
      browserHistory.push({ pathname: '/search/', query: { q: searchTerm } })

      dispatch(resetForm('search', { searchTerm: '' }))
    }
  }
}
