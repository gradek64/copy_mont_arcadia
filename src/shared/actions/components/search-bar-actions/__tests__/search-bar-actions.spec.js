import * as actions from '../search-bar-actions'

const snapshot = (action) => expect(action).toMatchSnapshot()

describe('Search Bar Actions', () => {
  describe('toggleSearchBar()', () => {
    it('should return the correct action', () => {
      snapshot(actions.toggleSearchBar())
    })
  })

  describe('closeProductsSearchBar()', () => {
    it('should return the correct action', () => {
      snapshot(actions.closeProductsSearchBar())
    })
  })

  describe('trackSearchBarSelected()', () => {
    it('should return the correct action', () => {
      snapshot(actions.trackSearchBarSelected())
    })
  })
})
