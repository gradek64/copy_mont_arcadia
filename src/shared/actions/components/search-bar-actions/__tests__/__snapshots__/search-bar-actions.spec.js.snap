// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search Bar Actions closeProductsSearchBar() should return the correct action 1`] = `
Object {
  "type": "CLOSE_PRODUCTS_SEARCH_BAR",
}
`;

exports[`Search Bar Actions toggleSearchBar() should return the correct action 1`] = `
Object {
  "type": "TOGGLE_PRODUCTS_SEARCH_BAR",
}
`;

exports[`Search Bar Actions trackSearchBarSelected() should return the correct action 1`] = `
Object {
  "payload": Object {
    "action": "searchbar",
    "category": "searchSelected",
  },
  "type": "MONTY/ANALYTICS.SEND_CLICK_EVENT",
}
`;
